### PR TITLE
Improve APIs for creating/updating direct paging integrations 

### PR DESCRIPTION
**Pull request by vadimkerr**
# What this PR does

- Disallow creating direct paging integrations for a team that already has one + make sure this constraint is also enforced on update
- Refactor some direct paging API parts to be more reusable & simple
- Add tests

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/public_api/serializers/integrations.py [hunk 1]
```diff
@@ -125,11 +125,14 @@ def create(self, validated_data):
...
-                organization=organization,
-            )
+            try:
+                instance = AlertReceiveChannel.create(
+                    **validated_data,
+                    author=self.context["request"].user,
+                    organization=organization,
+                )
+            except AlertReceiveChannel.DuplicateDirectPaging:
+                raise DuplicateDirectPagingBadRequest
```
Comment (vadimkerr) [inline engine/apps/public_api/serializers/integrations.py:135]: Making sure that direct paging integrations are unique per team in public API.

**Review by matiasb — APPROVED** [review]
LGTM

File: engine/apps/api/serializers/alert_receive_channel.py [hunk 2]
```diff
@@ -121,15 +121,24 @@ def create(self, validated_data):
...
-        )
+        try:
+            instance = AlertReceiveChannel.create(
+                **validated_data,
+                organization=organization,
+                author=self.context["request"].user,
+                allow_source_based_resolving=is_able_to_autoresolve,
+            )
+        except AlertReceiveChannel.DuplicateDirectPaging:
+            raise DuplicateDirectPagingBadRequest
```
[the remark this answers is not in the document]
Comment (iskhakov) [reply engine/apps/api/serializers/alert_receive_channel.py:132]: I would suggest keep `api_helpers` for more generic exceptions

May be use the same format as everywhere, until we have a decision
```
raise BadRequest(detail="Direct paging integration already exists for this team")
```
Or 
```
raise BadRequest(detail=str(AlertReceiveChannel.DuplicateDirectPaging))
```

File: engine/apps/alerts/models/alert_receive_channel.py [hunk 3]
```diff
@@ -251,6 +251,25 @@ def delete(self):
     def hard_delete(self):
         super(AlertReceiveChannel, self).delete()
 
+    class DuplicateDirectPaging(Exception):
```
[the remark this answers is not in the document]
Comment (vadimkerr) [reply engine/apps/alerts/models/alert_receive_channel.py]: Good call! Addressed in https://github.com/grafana/oncall/pull/2603/commits/06ed04dbb29f42ba4565d5def4e7bcf380bc1cf7

File: engine/apps/api/serializers/alert_receive_channel.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vadimkerr) [reply engine/apps/api/serializers/alert_receive_channel.py]: Great suggestions! Addressed in https://github.com/grafana/oncall/pull/2603/commits/06ed04dbb29f42ba4565d5def4e7bcf380bc1cf7

**Pull request merged**